src/llv8.cc

```diff
@@ -1311,6 +1311,86 @@ void JSObject::Keys(std::vector<std::string>& keys, Error& err) {
   return;
 }
 
+
+void JSObject::Entries(std::vector<std::pair<Value, Value>>& entries,
+                       Error& err) {
+  entries.clear();
+
+  HeapObject map_obj = GetMap(err);
+
+  Map map(map_obj);
+
+  bool is_dict = map.IsDictionary(err);
+  if (err.Fail()) return;
+
+  if (is_dict) {
+    DictionaryEntries(entries, err);
+  } else {
+    DescriptorEntries(entries, map, err);
+  }
+
+  return;
+}
+
+void JSObject::DictionaryEntries(std::vector<std::pair<Value, Value>>& entries,
```

Comment:
No, the performance issues were due to creating JSStrings when they weren't actually needed. That was fairly catastrophic. (I was creating a std::string out of every property name rather than just the ones that pointed to the value we were interested in.)

I made the assumption that it would be better to pass by ref but it looks like that was wrong. I did some basic wall clock timing of the two approaches on the core I used to re-create the original problem they actually both came out the same so I'll push the change.

Either copying the vector isn't that big a deal, the compiler is optimising it or the vector implementation is a bit clever about how it's copied. (Or any combination of those!)